### ISSUE TITLE
GraphQLParser.AST.GraphQLTypeExtensionDefinition was not handled

### ISSUE THREAD

**Issue by wingsking** [graphql-dotnet/graphql-dotnet#632]
```
type Query {
       message(id:String): Message
}

extend type Query {
       comment(id:String): Comment
}
```
If you try to define something like above with "extend type", the schema will not be built correctly. Looking at the code in SchemaBuilder.BuildSchemaFrom(), it doesn't handle the "GraphQLTypeExtensionDefinition" definition type. 

Another thing is that with current implementation, if you define a type multiple times, whichever comes last will win, which doesn't look right to me. How about throwing an exception if duplicate types are defined.

Would you be able to fix this?

**Comment by joemcbride** [comment]
Hey, thanks for trying out the project!  Those would certainly be things that can be added though I won't be able to get to them in the near future.  I would gladly review a PR for it.

**Comment by joemcbride** [comment]
Merged in #635
